docs/labs/sql-injection.html

```diff
@@ -0,0 +1,155 @@
+<!DOCTYPE html>
+<html>
+<head>
+<meta http-equiv="X-UA-Compatible" content="IE=edge">
+<meta name="viewport" content="width=device-width, initial-scale=1">
+<link rel="stylesheet" href="https://best.openssf.org/assets/css/style.css">
+<link rel="stylesheet" href="checker.css">
+<script src="js-yaml.min.js"></script>
+<script src="checker.js"></script>
+<link rel="license" href="https://creativecommons.org/licenses/by/4.0/">
+
+<!-- See create_labs.md for how to create your own lab! -->
+
+<!-- Sample expected answer -->
+<script id="expected0" type="plain/text">String QueryString = "select * from authors where lastname = ?";</script>
+
+<script id="correct0" type="plain/text">
+ (String QueryString)\s\=\s\"select\s+\*\s+from\s+authors\s+where\s+lastname\s\=\s+\?\"\;
```

Comment:
This "correct" pattern is too strict. For example, requires *exactly* one space around "=", but in reality, 0 or more are allowed. On the other hand, there must be at least one space between "String" and "QueryString" - in our system a space means "0 or more spaces allowed here".

So I'd rewrite this:
~~~~
 (String QueryString)\s\=\s\"select\s+\*\s+from\s+authors\s+where\s+lastname\s\=\s+\?\"\;
~~~~

Into:

~~~~
 String\s+QueryString = "select\s+\*\s+from\s+authors\s+where\s+lastname\s\=\s+\?\" \;
~~~~

It might be best to merge the two textareas into a single question.

Reply:
If you don't merge the two textareas, you might put a "// comment" above the new area to explain what you're doing.